### PR TITLE
Add Accounts.onLogout(func) Hook

### PR DESCRIPTION
**Pull request by pmoons**
`Accounts.onLogout(func)` is practically identical to the implementation of `Accounts.onLogin(func)`.  This PR is an effort to maintain symmetry between the available Login / Logout hooks.

Use cases include destroying active session variables, reverting UI elements, and any other cleanup that needs to occur immediately after the user is logged out.  

This PR is related to the following discussions:
[#5984](https://github.com/meteor/meteor/pull/5894)
#3867 
[#1074](https://github.com/meteor/meteor/issues/1074) 
[#426](https://github.com/meteor/meteor/issues/426)

Corresponding user-facing documentation has also been provided in this PR, as well as appropriate tests.

### CODE REVIEW

**Comment by pmoons** [comment]
Test failed in livedata package.  Could someone help me discover the issue on this please?  Only did work in `accounts-base` package.

Failed Test: 
```
tinytest - livedata stub - reconnect non-idempotent method : !!!!!!!!! FAIL !!!!!!!!!!!
{"groupPath":["tinytest","livedata stub"],"test":"reconnect non-idempotent method","events":[{"sequence":2489,"type":"fail","details":{"type":"true","not":false},"cookie":{"name":"livedata stub - reconnect non-idempotent method","offset":0,"groupPath":["tinytest","livedata stub"],"shortName":"reconnect non-idempotent method"}}]}
```

**Comment by jremi** [comment]
This feature is very important. Thank you for working on this, hopefully this onLogout hook will be available soon.

**Comment by tmeasday** [comment]
Can you rebase this to be off of current `devel`? I think this branch is off an earlier commit with failing self tests. It should be as easy as (assuming you have a remote called `upstream` pointing here):

```
# optionally
git remote add upstream https://github.com/meteor/meteor

git fetch upstream
git rebase upstream/devel
git push --force
```

Also, note that the docs have moved to https://github.com/meteor/docs. Sorry this PR got caught up in that change, so for now just remove the docs changes from the PR and we'll open a PR there once this commit is in.

**Comment by jdmswong** [comment]
add this!!

**Comment by tmeasday** [comment]
@pmoons?

**Comment by Ben305** [comment]
This would be great 👍

File: packages/accounts-password/password_tests_setup.js [hunk 1]
```diff
@@ -88,6 +96,14 @@ Accounts.onLoginFailure(function (attempt) {
   }
 });
 
+var capturedLogouts = {};
```
Comment (benjamn) [inline packages/accounts-password/password_tests_setup.js]: Shouldn't this be an array (`[]`)?

**Comment by tmeasday** [comment]
Ok, I rebased and squashed some commits (and addressed @benjamn's comment) here: https://github.com/meteor/meteor/pull/7187

**Comment by convexset** [comment]
Seems to be a major asymmetry here. `Accounts.onLogin` callbacks are called with an object argument with "signature" `{type, allowed, methodName, methodArguments, user, connection}`, and `Accounts.onLogout` callbacks are called with no arguments, and with no special function invocation context (`this`)...

The relevant connection id has to be scrounged up with `DDP._CurrentInvocation.getOrNullIfOutsideFiber()`, which strike me as... excessive. Whyyyyyyyyyyyyy?

**Comment by tmeasday** [comment]
@convexset https://github.com/meteor/meteor/pull/7433 should be released in 1.4.1 soon.

**Comment by interfaith** [comment]
is onLogout available ?
 after logging in with Twitter
a way to clean up before logging out is VITAL !!

**Pull request closed without merging**